lib/kubeclient.rb

```diff
@@ -8,9 +9,36 @@
 require 'kubeclient/replication_controller'
 require 'kubeclient/entity_list'
 require 'kubeclient/kube_exception'
-
+require 'kubeclient/watch'
 
 module Kubeclient
+
+  # HTTP Stream used to watch changes on entities
+  class WatchStream
```

Comment:
Many modules in the stdlib contain multiple classes per file. In rails instead it's discouraged/prohibited because it's breaking its own autoloading mechanism.

In our case I think it boils down to whether it belongs there or not logically: in my opinion it's part of the rest-api client engine that currently reside in that file.